### ISSUE TITLE
ancestries' ontology and /rest/ancestry_categories

### ISSUE THREAD

**Issue by ramiromagno** [PGScatalog/PGS_Catalog#185]
Hi PGS Catalog Team

Now that you have updated your ancestries' ontology by including the multi-ancestry categories, namely, *Multi-Ancestry
(including Europeans)* and *Multi-Ancestry (excluding Europeans)*, and also added the *display categories*, I would like to take the opportunity to ascertain if my assumptions are correct about how each relate to one another, and whether some renaming would be appropriate.

As I see it, we have now three levels of ancestry description (using my own wording here to refer to these levels):

1. **Ancestry** (`ancestry_free` in `Sample`). This corresponds to the most basic level of description. Examples include Chinese, Japanese, Korean, Spanish, Swedish, Brazilian, Mexican, etc.. Essentially, the concepts given as examples in the third column of [Table 1 of Morales et al.](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1396-2/tables/1).
2. **Ancestry category** (`ancestry_broad` in `Sample`). This is the ancestry category from the NHGRI-EBI GWAS Catalog framework, first column of [Table 1 of Morales et al.](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1396-2/tables/1), plus a couple of new categories, i.e., the multi-ancestry categories defined by you guys: *Multi-Ancestry (including Europeans)* and *Multi-Ancestry (excluding Europeans)*.
3. **Ancestry class**. This is a grouping of *ancestry categories* for visualisation purposes. You refer to this level by several names, such as: *Display category (for charts)* and *Display categories* in [/docs/ancestry/](https://www.pgscatalog.org/docs/ancestry/) and *Sample Ancestry* in [/docs/#desc_smpl](https://www.pgscatalog.org/docs/#desc_smpl).

As you can see from my bullet point no. 3, I am planning to refer to *display categor(y/ies)* as *ancestry class(es)* in quincunx. The reasons are:
- the term *display categories* has categories in it, which, in my opinion, might give the wrong impression that these are on the same hierarchical level as *ancestry categories*;
- having the word *display* reflects your original intention of using these new groupings for more effective visualisation in the web GUI, but I think it'd be better if the term were agnostic of its original end application;
- finally, the word *class* is already used to mean a group of categories in the context of Thematic Framework analysis, so it would be a nice co-opt.

Regarding the `/rest/ancestry_categories` that provides the mapping of ancestry symbols and their name, I think there are two issues here: 

- the mapping is actually between ancestry class symbols and ancestry class names, not between ancestry category symbols and ancestry category names, so the endpoint name might be a bit of a misnomer; but again, I know you're not calling it classes...
- the ancestry class names returned by `/rest/ancestry_categories` are in some cases simplified, i.e., not including the terms in brackets, e.g., `"Additional Asian Ancestries"` instead of `"Additional Asian Ancestries (including Central, and South East Asian)"`. This is fine if all one is doing is consultation, but for programmatic analysis, where these values might be matched across the database, this can be a problem.

I think it would be more useful if a full table of all these ancestries were returned. Currently, in quincunx, I have a saved dataset of ancestries that simultaneously provides the mapping between *ancestry categories* and *classes*, the mapping between *ancestry classes* and their *symbol* (as provided by `/rest/ancestry_categories`), the hexadecimal colour code of the class, and the *ancestry category* definition as given in your documentation. Perhaps you could provide these data in `/rest/ancestry_categories` instead?

```R
# A tibble: 19 x 6
   ancestry_category                       ancestry_class                           ancestry_class_sy… ancestry_class_co… definition                                                                                                                        examples            
   <chr>                                   <chr>                                    <chr>              <chr>              <chr>                                                                                                                             <chr>               
 1 Aboriginal Australian                   Additional Diverse Ancestries            OTH                #999999            "Includes individuals who either self-report or have been described by authors as Australian Aboriginal. These are expected to b… Martu Australian Ab…
 2 African American or Afro-Caribbean      African                                  AFR                #FFD900            "Includes individuals who either self-report or have been described by authors as African American or Afro-Caribbean. This categ… African American, A…
 3 African unspecified                     African                                  AFR                #FFD900            "Includes individuals that either self-report or have been described as African, but there was not sufficient information to all… African, non-Hispan…
 4 Asian unspecified                       Additional Asian Ancestries (including … ASN                #B15928            "Includes individuals that either self-report or have been described as Asian but there was not sufficient information to allow … Asian, Asian Americ…
 5 Central Asian                           Additional Asian Ancestries (including … ASN                #B15928            "Includes individuals who either self-report or have been described by authors as Central Asian. We note that there does not app… Silk Road (founder/…
 6 East Asian                              East Asian                               EAS                #4DAF4A            "Includes individuals who either self-report or have been described by authors as East Asian or one of the sub-populations from … Chinese, Japanese, …
 7 European                                European                                 EUR                #377EB8            "Includes individuals who either self-report or have been described by authors as European, Caucasian, white, or one of the sub-… Spanish, Swedish    
 8 Greater Middle Eastern (Middle Eastern… Greater Middle Eastern (Middle Eastern,… GME                #00CED1            "Includes individuals who self-report or were described by authors as Middle Eastern, North African, Persian, or one of the sub-… Tunisian, Arab, Ira…
 9 Hispanic or Latin American              Hispanic or Latin American               AMR                #E41A1C            "Includes individuals who either self-report or are described by authors as Hispanic, Latino, Latin American, or one of the sub-… Brazilian, Mexican  
10 Native American                         Additional Diverse Ancestries            OTH                #999999            "Includes indigenous individuals of North, Central, and South America, descended from the original human migration into the Amer… Pima Indian, Plains…
11 Not reported                            Ancestry Not Reported                    NR                 #BBBBBB            "Includes individuals for which no ancestry or country of recruitment information is available"                                   NA                  
12 Oceanian                                Additional Diverse Ancestries            OTH                #999999            "Includes individuals that either self-report or have been described by authors as Oceanian or one of the sub-populations from t… Solomon Islander, M…
13 Other                                   Additional Diverse Ancestries            OTH                #999999            "Includes individuals where an ancestry descriptor is known but insufficient information is available to allow assignment to one… Surinamese, Russian 
14 Other admixed ancestry                  Additional Diverse Ancestries            OTH                #999999            "Includes individuals who either self-report or have been described by authors as admixed and do not fit the definition of the o… NA                  
15 South Asian                             South Asian                              SAS                #984EA3            "Includes individuals who either self-report or have been described by authors as South Asian or one of the sub-populations from… Bangladeshi, Sri La…
16 South East Asian                        Additional Asian Ancestries (including … ASN                #B15928            "Includes individuals who either self-report or have been described by authors as South East Asian or one of the sub-populations… Thai, Malay         
17 Sub-Saharan African                     African                                  AFR                #FFD900            "Includes individuals who either self-report or have been described by authors as Sub-Saharan African or one of the sub-populati… Yoruban, Gambian    
18 Multi-Ancestry (including Europeans)    Multi-Ancestry (including Europeans)     MAE                #A6CEE3            "Combined sample of multiple ancestries that includes European ancestry individuals. Used when ancestry-specific sample sizes ar… NA                  
19 Multi-Ancestry (excluding Europeans)    Multi-Ancestry (excluding Europeans)     MAO                #FF7F00            "Combined sample of multiple ancestries that does not include any European ancestry individuals. Used when ancestry-specific sam… NA
```

I am happy to hear your thoughts on this.

**Comment by ens-lgil** [comment]
Dear @ramiromagno,

When we implemented the "Display category", at the beginning this was only meant to be used in the pie charts, to provide a **summarised** information about the ancestry distribution for each Score.
We also updated the "ancestry_broad" data so now they all match the NHGRI-EBI GWAS ancestry categories.

Then when we did some users testing, it appeared that there was an interest to have these distributions available in the metadata exports and, by extend, in the REST API. However we didn't intend to provide a detailed mapping of the distribution with the "Ancestry categories".

Regarding the `rest/score` endpoints, we choose to use a 2/3 letters notation to reduce the size of the rest call (mainly for the `rest/score/all` endpoint) and keep the endpoints as fast as we could (we store the distribution in the database in a similar structure, therefore this is relatively quick to build and return the REST API call).

The `/rest/ancestry_categories` endpoint is mainly there to provide the labels corresponding to the 2/3 letters ancestry notations.
However, thanks to your suggestions, we are thinking about improving the `/rest/ancestry_categories` endpoint by providing the "Ancestry categories" associated with the "Display categories", e.g.:
```
"AFR": {
	"display_category": "African", 
	"categories": [
		"African American or Afro-Caribbean", 
		"African unspecified",
		"Sub-Saharan African"
	]
},
"EUR": {
	"display_category": "European",
	"categories": [
		"European"
	]
},
...
```

Best regards,
Laurent
